gulpfile.js

```diff
@@ -0,0 +1,106 @@
+"use strict";
+
+var gulp = require("gulp");
+var beautifyJs = require("./build_tools/beautifyJs");
+var lintJs = require("./build_tools/lintJs");
+var bundle = require("./build_tools/bundle");
+var sequence = require("./build_tools/sequence");
+var clientSideJasmine = require("./build_tools/clientSideJasmine");
+var serverSideJasmine = require("./build_tools/serverSideJasmine");
+var benchmark = require("./build_tools/benchmark");
+
+var LIB = "./lib/**/*.js";
+var REST = "./*.js";
+var BUILD_TOOLS = "./build_tools/**/*.js";
+var TEST_HELPERS = "./spec/helpers/**/*.js";
+var CLIENT_TESTS = "./spec/client/**/*.js";
+var CLIENT_TEST_BUNDLE = "./spec/build/lib.js";
+var CONFIGURE_CLIENT_TEST = "./spec/configureClientTesting.js";
+var SERVER_TESTS = "./spec/server/**/*.js";
+var CONFIGURE_SERVER_TEST = "./spec/configureServerTesting.js";
+var BENCHMARKS = "./benchmarks/**/*.js";
+var ALL_TEST_CODE = "./spec/**/*.js";
+var NOT_BUILD_DIRECTORY_THOUGH = "!./spec/build/**/*.js";
+
+var ALL_CODE = [
+    BUILD_TOOLS,
+    LIB,
+    REST,
+    ALL_TEST_CODE,
+    NOT_BUILD_DIRECTORY_THOUGH
+];
+
+gulp.task("beautifyJs", function() {
+    return beautifyJs(ALL_CODE);
+});
+
+gulp.task("lintJs", function() {
+    return lintJs(ALL_CODE);
+});
+
+gulp.task("test-on-server", function() {
+    return serverSideJasmine([
+        CONFIGURE_SERVER_TEST,
+        TEST_HELPERS,
+        SERVER_TESTS
+    ]);
+});
+
+gulp.task("benchmarks", function() {
+    return benchmark(BENCHMARKS);
+});
+
+gulp.task("bundle-for-client", function() {
+    return bundle({
+        src: LIB,
+        dest: CLIENT_TEST_BUNDLE,
+        alias: {
+            "backbone": "./node_modules/backbone/backbone.js",
```

Comment:
A comment for another time, but are we doing something wrong with jasmine that it requires these aliases?